src/databricks/labs/ucx/hive_metastore/views_sequencer.py

```diff
@@ -54,13 +56,17 @@ def _compute_dependencies(self, all_tables: dict[str, Table]):
     def _parse_view_text(self) -> SqlExpression:
         try:
             # below can never happen but avoids a pylint error
-            assert self._view.view_text is not None
-            statements = sqlglot.parse(self._view.view_text)
+            assert self._view.src.view_text is not None
+            statements = sqlglot.parse(self._view.src.view_text)
             if len(statements) != 1 or statements[0] is None:
-                raise ValueError(f"Could not analyze view SQL: {self._view.view_text} of table {self._view.key}")
+                raise ValueError(
+                    f"Could not analyze view SQL: {self._view.src.view_text} of table {self._view.src.key}"
```

Comment:
We could _not_ halt if no other view uses that view. But we really don't know that at this point.
Is it reasonable to assume the following ?
 - view_text is valid SQL (hence the above fragment is just defensive coding)
 - if the assessment or the migration fail due to a schema change, then rerunning the whole thing would self-heal
 - if it doesn't then human intervention is likely necessary

Reply:
> view_text is valid SQL (hence the above fragment is just defensive coding)

correct. it's a valid SQL. but we need to pre-empt the defence from `sqlglot` bugs.

the other two assumptions are correct

Reply:
> pre-empt the defence from sqlglot bugs

can you clarify what you mean exactly

Reply:
when `sqlglot` has bugs parsing Databricks queries, we should skip only those queries, log an error, and proceed with the rest of the queries. e.g. if we have 100000 queries and a third-party library bug affects only 2-3 of those, we skip those 2-3 queries and migrate 99997 queries.

Reply:
Mmm... aren't queries and views different animals ? If we can't migrate a view due to its failing query (or any other cause), then shouldn't the migration of all subsequent views or queries that rely on that view be skipped ? 
I'll give you that the current code doesn't skip just these, but isn't it better to interrupt the whole process rather than create an inconsistent state (or more accurately, since the migration of depending objects will fail, a possibly large -and scary- bunch of errors).
Isn't it better to tell the user: "Hey, there seems to be a problem _with this view_, can you check it and re-run the migration process" rather than throw at them possibly dozens of errors ?
If the former, we can improve the code such that we skip only views that are bound to fail migration. Would that be high priority over other ast issues (#1082 or #1109)